App.tsx

```diff
@@ -1,20 +1,48 @@
-import {ApolloClient, ApolloProvider, InMemoryCache} from '@apollo/client';
+import {
+  ApolloClient,
+  ApolloLink,
+  ApolloProvider,
+  concat,
+  HttpLink,
+  InMemoryCache,
+} from '@apollo/client';
+import AsyncStorage from '@react-native-async-storage/async-storage';
 import React from 'react';
 import {Navigation} from 'react-native-navigation';
 import {HomePage} from './src/screens/home/home-page';
 import {LoginPage} from './src/screens/login/login-page';
+import {UserPage} from './src/screens/users/user-page';
 
-const client = new ApolloClient({
+const httpLink = new HttpLink({
   uri: 'https://tq-template-server-sample.herokuapp.com/graphql',
+});
+
+const auth = new ApolloLink((operation, forward) => {
+  const token = AsyncStorage.getItem('token');
```

Comment:
hmmm aqui deveria ser 
```suggestion
  const token = await AsyncStorage.getItem('token');
```
do jeito que está, o token está sendo lido corretamente? 🤔